src/main/java/teammates/storage/search/SearchManager.java

```diff
@@ -1,169 +1,99 @@
 package teammates.storage.search;
 
-import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 
-import com.google.appengine.api.search.Document;
-import com.google.appengine.api.search.Index;
-import com.google.appengine.api.search.IndexSpec;
-import com.google.appengine.api.search.OperationResult;
-import com.google.appengine.api.search.PutException;
-import com.google.appengine.api.search.PutResponse;
-import com.google.appengine.api.search.Query;
-import com.google.appengine.api.search.Results;
-import com.google.appengine.api.search.ScoredDocument;
-import com.google.appengine.api.search.SearchServiceFactory;
-import com.google.appengine.api.search.StatusCode;
-
-import teammates.common.exception.TeammatesException;
+import teammates.common.datatransfer.InstructorSearchResultBundle;
+import teammates.common.datatransfer.StudentSearchResultBundle;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.SearchNotImplementedException;
 import teammates.common.util.Logger;
-import teammates.common.util.retry.MaximumRetriesExceededException;
-import teammates.common.util.retry.RetryManager;
-import teammates.common.util.retry.RetryableTaskThrows;
+import teammates.common.util.StringHelper;
 
 /**
- * Manages {@link Document} and {@link Index} in the Datastore for use of search functions.
- *
- * @see <a href="https://cloud.google.com/appengine/docs/java/search/">https://cloud.google.com/appengine/docs/java/search/</a>
+ * Acts as a proxy to search service.
  */
 public final class SearchManager {
 
-    private static final String ERROR_NON_TRANSIENT_BACKEND_ISSUE =
-            "Failed to put document(s) %s into search index %s due to non-transient backend issue: ";
-    private static final String ERROR_MAXIMUM_RETRIES_EXCEEDED =
-            "Failed to put document(s) %s into search index %s after maximum retries: %s: ";
     private static final Logger log = Logger.getLogger();
-    private static final ThreadLocal<Map<String, Index>> PER_THREAD_INDICES_TABLE = new ThreadLocal<>();
 
-    private static final RetryManager RM = new RetryManager(8);
+    private String searchServiceHost;
+
+    public SearchManager(String searchServiceHost) {
+        this.searchServiceHost = searchServiceHost;
+    }
 
-    private SearchManager() {
-        // utility class
+    private boolean isSearchServiceActive() {
+        return !StringHelper.isEmpty(searchServiceHost);
     }
```

Comment:
Contrary to the `TestProperties` counterpart, return value of this function will always be negated.
How about `isSearchServiceInactive()`?

Reply:
@Derek-Hardy I'll leave this to later edit, if absolutely necessary